### ISSUE TITLE
Suggestion: document a simpler CMake integration via FetchContent

### ISSUE THREAD

**Issue by gershnik** [doctest/doctest#823]
Currently in [build-systems.md](https://github.com/doctest/doctest/blob/master/doc/markdown/build-systems.md#cmake) the suggested CMake integrations are either via manual file download or `ExternalProject`.
The first is, well, manual and the second is heavyweight, cumbersome and downloads at build time (yikes!).

There is a much simpler way to integrate doctest using [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html).

Something like this:

```cmake
include(FetchContent)

FetchContent_Declare(doctest
    URL  https://raw.githubusercontent.com/doctest/doctest/v2.4.11/doctest/doctest.h
    DOWNLOAD_NO_EXTRACT TRUE
    DOWNLOAD_EXTRACT_TIMESTAMP TRUE #for CMake 3.24 and higher
)
FetchContent_MakeAvailable(doctest)

set(DOCTEST_INCLUDE_DIR ${doctest_SOURCE_DIR})
```

This downloads just the header and does it during configure, not build, time.

If you think this is reasonable to document in addition or instead of `ExternalProject` method happy to submit a PR.

**Comment by onqtam** [comment]
https://github.com/doctest/doctest/issues/698
https://github.com/doctest/doctest/pull/821

**Comment by gershnik** [comment]
Ah, my issue search skills are obviously pretty bad, thank you! 
However, the approach suggested in #698 and implemented in #821 is still suboptimal for doctest unless one needs `doctest_discover_tests(<target>)` functionality.
It works but it clones the GitHub repo and perform CMake configure on it. The library is a single header and none of it is really necessary.
The approach above is bare-bones and faster - only a single header is downloaded and no configuration is performed at all.
 
Thank you!

**Comment by cdeln** [comment]
@gershnik Do you want to add this approach to the docs?

**Comment by gershnik** [comment]
@cdeln here you go: #866

**Comment by gershnik** [comment]
@balping Indeed. I have a vague recollection that older CMakes emitted a warning *without* it but the ones I have now reject it.
The correct code is:
```cmake
FetchContent_Declare(doctest
    URL  https://raw.githubusercontent.com/doctest/doctest/v2.4.11/doctest/doctest.h
    DOWNLOAD_NO_EXTRACT TRUE
)
```

**Comment by mitchgrout** [comment]
Closed by #866
